x-pack/plugins/ml/server/models/data_recognizer/modules/apm_transaction/ml/datafeed_apm_metrics.json

```diff
@@ -0,0 +1,87 @@
+{
+  "job_id": "JOB_ID",
+  "indices": [
+    "INDEX_PATTERN_NAME"
+  ],
+  "chunking_config" : {
+    "mode" : "off"
+  },
+  "aggregations" : {
+    "buckets" : {
+      "composite" : {
+        "size" : 5000,
+        "sources" : [
+          {
+            "date" : {
+              "date_histogram" : {
+                "field" : "@timestamp",
+                "fixed_interval" : "90s"
+              }
+            }
+          },
+          {
+            "transaction.type" : {
+              "terms" : {
+                "field" : "transaction.type"
+              }
+            }
+          },
+          {
+            "service.name" : {
+              "terms" : {
+                "field" : "service.name"
+              }
+            }
+          }
+        ]
+      },
+      "aggs" : {
+        "@timestamp" : {
+          "max" : {
+            "field" : "@timestamp"
+          }
+        },
+        "transactions_per_min" : {
+          "rate" : {
+            "unit" : "minute"
```

Comment:
I'm getting an "all shards failed" error in the datafeed. If I run this search in dev tools, ES complains about the rate aggregation not being nested in a date histo agg:

```
{
  "error" : {
    "root_cause" : [
      {
        "type" : "illegal_argument_exception",
        "reason" : "The rate aggregation can only be used inside a date histogram"
      },
      {
        "type" : "illegal_argument_exception",
        "reason" : "The rate aggregation can only be used inside a date histogram"
      }
    ],
    "type" : "search_phase_execution_exception",
    "reason" : "all shards failed",
    "phase" : "query",
    "grouped" : true,
    "failed_shards" : [
      {
        "shard" : 0,
        "index" : "apm-7.15.0-metric-000001",
        "node" : "i8CJFYgaRDGDcmj5y1d0AA",
        "reason" : {
          "type" : "illegal_argument_exception",
          "reason" : "The rate aggregation can only be used inside a date histogram"
        }
      },
      {
        "shard" : 0,
        "index" : "apm-7.15.0-metric-2021.08.30-000001",
        "node" : "QucFwja8Q9mB9jbmzvl8aQ",
        "reason" : {
          "type" : "illegal_argument_exception",
          "reason" : "The rate aggregation can only be used inside a date histogram"
        }
      }
    ],
    "caused_by" : {
      "type" : "illegal_argument_exception",
      "reason" : "The rate aggregation can only be used inside a date histogram",
      "caused_by" : {
        "type" : "illegal_argument_exception",
        "reason" : "The rate aggregation can only be used inside a date histogram"
      }
    }
  },
  "status" : 400
}
```

Reply:
@dgieselaar its only supported in 7.16+

Reply:
Will this rate agg run on documents with custom `_doc_count`? If that's the case it's currently broken and we don't have confirmation that this will be fixed for 7.16.